src/metadataBlocks/infra/repositories/transformers/metadataBlockTransformers.ts

```diff
@@ -51,3 +56,28 @@ const transformPayloadMetadataFieldInfo = (
   }
   return metadataFieldInfo
 }
+
+/**
+ * This method checks if a new metadata field key is already present in the metadata fields.
+ * We need this method since child fields are returned as sub-objects and as root objects in the response payload, so we don't want to replicate this behavior in the model.
+ *
+ * @param {Record<string, MetadataFieldInfo>} [metadataFields] - The current transformed metadata fields
+ * @param {string} newMetadataFieldKey - The new metadata field key to transform
+ *
+ * @returns {boolean}
+ */
+const metadataFieldIsAlreadyPresentAsChildField = (
+  metadataFields: Record<string, MetadataFieldInfo>,
+  newMetadataFieldKey: string
+): boolean => {
+  let isPresent = false
+  Object.keys(metadataFields).forEach((metadataFieldKey) => {
+    if (
+      metadataFields[metadataFieldKey].childMetadataFields &&
+      metadataFields[metadataFieldKey].childMetadataFields[newMetadataFieldKey]
+    ) {
+      isPresent = true
+    }
+  })
+  return isPresent
+}
```

Comment:
While it might not make a significant difference, I noticed something that could easily improve the performance of these loops

I was thinking that if we collect all keys that exist as child fields into a set before the main loop begins, we could avoid some unnecessary iterations.

1. First, collect all `childFieldKeys`:

```javascript
const childFieldKeys = new Set();
Object.values(metadataBlockFieldsPayload).forEach((fieldInfo) => {
  if (fieldInfo.childFields) {
    Object.keys(fieldInfo.childFields).forEach(childKey => {
      childFieldKeys.add(childKey);
    });
  }
});
```

2. Then, execute the main loop:

```javascript
Object.keys(metadataBlockFieldsPayload).forEach((metadataFieldKey) => {
  if (!childFieldKeys.has(metadataFieldKey)) { // Check against the precomputed set for efficiency
    const metadataFieldInfoPayload = metadataBlockFieldsPayload[metadataFieldKey];
    metadataFields[metadataFieldKey] = transformPayloadMetadataFieldInfo(metadataFieldInfoPayload);
  }
});
```

Originally, we were checking if a field is a child field by iterating over all fields, which has a time complexity of O(n²). By storing the `childKeys` in a set, we reduce the time complexity of this operation to O(1), thereby improving efficiency